dashboard/src/components/AppView/AppControls/RollbackButton/RollbackButton.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from "react";
+import * as Modal from "react-modal";
+import { IAppRepository, IChartVersion, IRelease } from "shared/types";
+import SelectRepoForm from "../../../../components/SelectRepoForm";
+import RollbackDialog from "./RollbackDialog";
+
+interface IRollbackButtonProps {
+  app: IRelease;
+  rollbackApp: (
+    chartVersion: IChartVersion,
+    releaseName: string,
+    revision: number,
+    namespace: string,
+    values?: string,
+  ) => Promise<boolean>;
+  chart?: IChartVersion;
+  getChartVersion: (id: string, version: string) => Promise<void>;
+  loading: boolean;
+  repos: IAppRepository[];
+  repo: IAppRepository;
+  kubeappsNamespace: string;
+  fetchRepositories: () => void;
+  checkChart: (repo: string, chartName: string) => Promise<boolean>;
+  error?: Error;
+}
+
+interface IRollbackButtonState {
+  modalIsOpen: boolean;
+  loading: boolean;
+  chartName: string;
+  chartVersion: string;
+}
+
+class RollbackButton extends React.Component<IRollbackButtonProps> {
+  public state: IRollbackButtonState = {
+    modalIsOpen: false,
+    loading: false,
+    chartName: "",
+    chartVersion: "",
+  };
+
+  public componentWillReceiveProps(nextProps: IRollbackButtonProps) {
```

Comment:
[This method is deprecated](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops) and it will be removed in future React versions. Since we're in `16.8.6`, we should stop adding more deprecated methods to the codebase.

For this use case, you can use the [getDerivedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) lifecycle method.